ludwig/data/dataset/tfrecord.py

```diff
@@ -190,6 +190,7 @@ def create(self, dataset, config, training_set_metadata):
         )
 
     def create_inference_dataset(self, dataset, tag, config, training_set_metadata):
+        # TODO(shreya): Confirm if this needs to be updated to RayDatasets too.
```

Comment:
Probably we want to remove PartitionedDataset here and just raise an exception like:

```
raise ValueError('Batch inference not supported with TFRecord format at this time')
```